float64 totalUT
string[] next_node
string[] assigned_edge